system/lib/standalone/paths.c

```diff
@@ -0,0 +1,233 @@
+#define _GNU_SOURCE
+#include "paths.h"
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sysexits.h>
+
+/// A name and file descriptor pair.
```

Comment:
How much of the code in this file is actually derived from the wasi-libc code?  If its actually derived from that codebase we should probably say so in the licence comment at the top.

Does wasi-libc use the triple-slash (///) for comments?   If not I would probably stick to normal C comments.